handler/src/main/java/io/netty/handler/ssl/ApplicationProtocolNegotiationHandler.java

```diff
@@ -151,4 +165,24 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
         ctx.fireExceptionCaught(cause);
         ctx.close();
     }
+
+    /**
+     * This abstract class is used to handle initial messages until ALPN is configuring pipeline.
+     * Once ALPN pipeline is successfully configured then it'll call {@link #success()}.
+     */
+    public static abstract class PipelineCallback extends ChannelInboundHandlerAdapter {
```

Comment:
I think it is early to expose this as a `public` API or even a different handler. Can we just have the `backlogList` inside the `ApplicationProtocolNegotiationHandler` itself?

Reply:
We can do but `ApplicationProtocolNegotiationHandler` does not remove itself from the pipeline even after ALPN is successful. Should I add a method to remove itself from the pipeline on a successful pipeline setup?